pynestml/src/main/grammars/org/PyNESTML.g4

```diff
@@ -0,0 +1,329 @@
+           | unitlessLiteral=NUMERIC_LITERAL divOp='/' right=unitType
+           | unit=NAME;
+
+  /*********************************************************************************************************************
+  * Expressions-Language
+  *********************************************************************************************************************/
+  /**
+   ASTExpr, i.e., several subexpressions combined by one or more
+   operators, e.g., 10mV + V_m - (V_reset * 2)/ms ....
+   or a simple expression, e.g. 10mV.
+  */
+  expression : leftParentheses='(' term=expression rightParentheses=')'
+         | <assoc=right> left=expression powOp='**' right=expression
+         | unaryOperator term=expression
+         | left=expression (timesOp='*' | divOp='/' | moduloOp='%') right=expression
+         | left=expression (plusOp='+'  | minusOp='-') right=expression
+         | left=expression bitOperator right=expression
+         | left=expression comparisonOperator right=expression
+         | logicalNot='not' term=expression
+         | left=expression logicalOperator right=expression
+         | condition=expression '?' ifTrue=expression ':' ifNot=expression
+         | simpleExpression
+         ;
+
+
+  /**
+    ASTSimpleExpression, consisting of a single element without combining operator, e.g.,
+    10mV, inf, V_m.
+  */
+  simpleExpression : functionCall
+                   | BOOLEAN_LITERAL // true & false;
+                   | NUMERIC_LITERAL (variable)?
+                   | NAME
+                   | isInf='inf'
+                   | variable;
+
+  unaryOperator : (unaryPlus='+' | unaryMinus='-' | unaryTilde='~');
+
+  bitOperator : (bitAnd='&'| bitXor='^' | bitOr='|' | bitShiftLeft='<<' | bitShiftRight='>>');
+
+  comparisonOperator : (lt='<' | le='<=' | eq='==' | ne='!=' | ne2='<>' | ge='>=' | gt='>');
+
+  logicalOperator : (logicalAnd='and' | logicalOr='or' );
+
+  /**
+    ASTVariable Provides a 'marker' AST node to identify variables used in expressions.
+    @attribute name
+  */
+  variable : NAME (differentialOrder)*;
+
+  /**
+    ASTFunctionCall Represents a function call, e.g. myFun("a", "b").
+    @attribute name The (qualified) name of the fucntions
+    @attribute args Comma separated list of expressions representing parameters.
+  */
+  functionCall : calleeName=NAME '(' (args=arguments)? ')';
+
+  arguments : expression (',' expression)*;
+
+
+  /*********************************************************************************************************************
+  * Equations-Language
+  *********************************************************************************************************************/
+  /** ASTOdeDeclaration. Represents a block of equations and differential equations.
+
+    @attribute Equation      List with equations, e.g. "I = exp(t)" od differential equations.
+  */
+  odeDeclaration  : (equation | shape | odeFunction | NEWLINE)+;
```

Comment:
We don't need this production. You can embed the rhs where the `odeDeclaration` was used.

Reply:
This part has been reworked according to your feedback.